### PR TITLE
[ALLUXIO-1718] Add integration tests for file free and delete

### PR DESCRIPTION
**Pull request by ifcharming**
https://tachyon.atlassian.net/browse/ALLUXIO-1718

This PR includes two integration tests:
1) ConcurrentFree in FileSystemMasterIntegrationTest
2) Client integration test for persisted file being freed and then deleted, verify the block metadata removal and check there's no FileDoesNotExistException.

### CODE REVIEW

**Comment by ifcharming** [comment]
@yupeng9

File: tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.WORKER_BLOCK_SYNC, 5,
+        TimeUnit.SECONDS));
+    HeartbeatScheduler.schedule(HeartbeatContext.WORKER_BLOCK_SYNC);
+    Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.WORKER_BLOCK_SYNC, 5,
+        TimeUnit.SECONDS));
+
+    status = mFileSystem.getStatus(filePath);
+    Assert.assertEquals(PersistenceState.PERSISTED.toString(), status.getPersistenceState());
+    // Block metadata is still present after block freed.
+    Assert.assertEquals(1, status.getBlockIds().size());
```
Comment (yupeng9) [inline tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java]: Can you also verify the block does have any locations?

File: tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.MASTER_LOST_FILES_DETECTION, 5,
+        TimeUnit.SECONDS));
+    HeartbeatScheduler.schedule(HeartbeatContext.MASTER_LOST_FILES_DETECTION);
+    Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.MASTER_LOST_FILES_DETECTION, 5,
+        TimeUnit.SECONDS));
+
+    // Verify the blocks are not in mLostBlocks.
+    BlockMaster bm = PrivateAccess.getBlockMaster(mLocalAlluxioClusterResource.get().getMaster()
+        .getInternalMaster());
+    Assert.assertEquals(0, bm.getLostBlocks().size());
```
Comment (yupeng9) [inline tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java]: Can you verify the block is lost after free before deletion?

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java [hunk 3]
```diff
@@ -119,6 +119,61 @@ public void exec(int depth, int concurrencyDepth, AlluxioURI path) throws Except
     }
   }
 
+  class ConcurrentFreer implements Callable<Void> {
```
Comment (yupeng9) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: Can you add comments about what this test does?

**Comment by yupeng9** [comment]
Thanks for adding the tests. I left some comments.

**Comment by yupeng9** [comment]
@ifcharming any updates?

File: tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java (hunk 1 above)
Comment (ifcharming) [inline tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java]: Done. Thanks!

File: tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java (hunk 2 above)
Comment (ifcharming) [inline tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java]: Added, thanks for the suggestion! Actually identified a problem in setting up testing environment by adding this check.

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java (hunk 3 above)
Comment (ifcharming) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: Done.

**Comment by ifcharming** [comment]
Thanks @yupeng9 , comments are addressed. PTAL.

**Comment by ifcharming** [comment]
Jenkins, test this please

**Comment by ifcharming** [comment]
Jenkins, test this please

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**